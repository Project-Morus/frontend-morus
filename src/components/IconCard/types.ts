
import { IBaseProps } from "../../types";
import { IconNamesTypes, StrokeType } from "../Icon/types";

export interface IIconCardProps extends IBaseProps {
  icon: IconNamesTypes,
  stroke?: StrokeType,
  variant?: VariantsCardTypes,
}

export type VariantsCardTypes = 'primary' | 'white'