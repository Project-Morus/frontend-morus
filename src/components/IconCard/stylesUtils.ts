export const SQUARE_DIMENSIONS = '70px';