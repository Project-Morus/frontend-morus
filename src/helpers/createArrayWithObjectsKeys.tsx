export const createArrayWithObjectsKeys = (quantity: number) => {
  const dotsKeys = []

  for (let i = 1; i <= quantity; i++) {
    [...dotsKeys, dotsKeys.push({ key: i })]
  }

  return dotsKeys
}