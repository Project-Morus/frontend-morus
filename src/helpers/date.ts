export const date = new Date()

export function formatDate(date: string) {
  const convertDate = new Date(date)

  return new Intl.DateTimeFormat('pt-BR').format(convertDate)
}

export function currentDateFormatted() {
  return new Intl.DateTimeFormat('pt-BR', { timeZone: 'UTC' }).format(date)
}