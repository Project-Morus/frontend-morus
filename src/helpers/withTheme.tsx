import { ComponentType, JSX } from 'react';
import { ThemeProvider } from "styled-components";
import { defaultTheme } from "../styles/themes/default";

type ComponentPropsType<T> = T & JSX.IntrinsicAttributes

export function withTheme<T>(WrapperComponent: ComponentType<ComponentPropsType<T>>) {
  return (props: ComponentPropsType<T>) => {
    return (
      <ThemeProvider theme={defaultTheme}>
        <WrapperComponent {...props} />
      </ThemeProvider>
    );
  };
}

