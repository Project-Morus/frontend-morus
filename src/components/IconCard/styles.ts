import styled from "styled-components";
import { SQUARE_DIMENSIONS } from "./stylesUtils";
import { VariantsCardTypes } from "./types";

export const CardSC = styled.div<{ $variant: VariantsCardTypes }>`
  width: ${SQUARE_DIMENSIONS};
  height: ${SQUARE_DIMENSIONS};

  display: flex;
  align-items: center;
  justify-content: center;

  background-color: ${({ theme, $variant }) => $variant === 'primary' ? theme.colors.primary[50] : theme.colors.white[50]};

  box-shadow: 0px 4px 4px 0px rgba(0, 0, 0, 0.05);

  border-radius: ${({ theme }) => theme.border.radius.regular};
`