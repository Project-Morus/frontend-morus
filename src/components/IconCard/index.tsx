import Icon from "../Icon";
import { CardSC } from "./styles";
import { IIconCardProps } from "./types";

const IconCard = ({ icon, stroke, variant = 'white', ...props }: IIconCardProps) => {

  return (
    <CardSC $variant={variant}
      role={props.role}
      className={props.className}
      data-testid={props?.['data-testid']}
    >
      <Icon
        name={icon}
        size="47"
        stroke={stroke}
      />
    </CardSC>
  );
}

export default IconCard;