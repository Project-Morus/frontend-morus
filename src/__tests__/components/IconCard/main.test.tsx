import { render, screen } from '@testing-library/react'
import { withTheme } from '../../../helpers/withTheme'
import IconCard from '../../../components/IconCard'

const IconCardWithTheme = withTheme(IconCard)

describe('IconCard component', () => {
  test('Verify if component is rendering', () => {
    render(<IconCardWithTheme icon="ph-smiley" role='roleIconCard' />)

    expect(screen.getByRole('roleIconCard')).toBeVisible()
  })

  test('Snapshot IconCard', () => {
    const { asFragment } = render(
      <IconCardWithTheme icon="ph-smiley" />,
    )
    expect(asFragment()).toMatchSnapshot()
  })

  test('Snapshot IconCard with variant primary', () => {
    const { asFragment } = render(
      <IconCardWithTheme icon="ph-smiley" variant='primary' />,
    )
    expect(asFragment()).toMatchSnapshot()
  })
})