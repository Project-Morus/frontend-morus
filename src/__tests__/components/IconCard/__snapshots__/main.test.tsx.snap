// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IconCard component Snapshot IconCard 1`] = `
<DocumentFragment>
  .c1 {
  color: #0353A4;
}

.c2 {
  display: flex;
  align-items: center;
  justify-content: center;
  max-width: 100%;
  width: fit-content;
  width: -moz-fit-content;
  height: fit-content;
  height: -moz-fit-content;
}

.c0 {
  width: 70px;
  height: 70px;
  display: flex;
  align-items: center;
  justify-content: center;
  background-color: 50:#FFF;
  100: #F8F8F8;
  box-shadow: 0px 4px 4px 0px rgba(0, 0, 0, 0.05);
  border-radius: 5px;
}

<div
    class="c0"
  >
    <div
      class="c1"
    >
      <div
        class="c2"
        data-testid="ph-smiley"
      >
        <svg
          fill="currentColor"
          height="47"
          viewBox="0 0 256 256"
          width="47"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M128,24A104,104,0,1,0,232,128,104.11,104.11,0,0,0,128,24Zm0,192a88,88,0,1,1,88-88A88.1,88.1,0,0,1,128,216ZM80,108a12,12,0,1,1,12,12A12,12,0,0,1,80,108Zm96,0a12,12,0,1,1-12-12A12,12,0,0,1,176,108Zm-1.07,48c-10.29,17.79-27.4,28-46.93,28s-36.63-10.2-46.92-28a8,8,0,1,1,13.84-8c7.47,12.91,19.21,20,33.08,20s25.61-7.1,33.07-20a8,8,0,0,1,13.86,8Z"
          />
        </svg>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`IconCard component Snapshot IconCard with variant primary 1`] = `
<DocumentFragment>
  .c1 {
  color: #0353A4;
}

.c2 {
  display: flex;
  align-items: center;
  justify-content: center;
  max-width: 100%;
  width: fit-content;
  width: -moz-fit-content;
  height: fit-content;
  height: -moz-fit-content;
}

.c0 {
  width: 70px;
  height: 70px;
  display: flex;
  align-items: center;
  justify-content: center;
  background-color: #E6EEF6;
  box-shadow: 0px 4px 4px 0px rgba(0, 0, 0, 0.05);
  border-radius: 5px;
}

<div
    class="c0"
  >
    <div
      class="c1"
    >
      <div
        class="c2"
        data-testid="ph-smiley"
      >
        <svg
          fill="currentColor"
          height="47"
          viewBox="0 0 256 256"
          width="47"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M128,24A104,104,0,1,0,232,128,104.11,104.11,0,0,0,128,24Zm0,192a88,88,0,1,1,88-88A88.1,88.1,0,0,1,128,216ZM80,108a12,12,0,1,1,12,12A12,12,0,0,1,80,108Zm96,0a12,12,0,1,1-12-12A12,12,0,0,1,176,108Zm-1.07,48c-10.29,17.79-27.4,28-46.93,28s-36.63-10.2-46.92-28a8,8,0,1,1,13.84-8c7.47,12.91,19.21,20,33.08,20s25.61-7.1,33.07-20a8,8,0,0,1,13.86,8Z"
          />
        </svg>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
